File: scala/src/main/scala/com/spoonacular/client/model/InlineResponse20030.scala
package com.spoonacular.client.model

import io.circe._
import io.finch.circe._
import io.circe.generic.semiauto._
import io.circe.java8.time._
import spoonacular._
import com.spoonacular.client.model.InlineResponse20028Nutrition
import com.spoonacular.client.model.InlineResponse20028Servings
import com.spoonacular.client.model.InlineResponse20030Ingredients
import java.math.BigDecimal
import scala.collection.immutable.Seq

/**
 * 
 * @param id 
 * @param title 
 * @param breadcrumbs 
 * @param imageType 
 * @param badges 
 * @param importantBadges 
 * @param ingredientCount 
 * @param generatedText 
 * @param ingredientList 
 * @param ingredients 
 * @param likes 
 * @param aisle 
 * @param nutrition 
 * @param price 
 * @param servings 
 * @param spoonacularScore 
 */
case class InlineResponse20030(id: Int,
                title: String,
                breadcrumbs: Seq[String],
                imageType: String,
                badges: Seq[String],
                importantBadges: Seq[String],
                ingredientCount: Int,
                generatedText: Option[Object],
                ingredientList: String,
                ingredients: Seq[InlineResponse20030Ingredients],
                likes: BigDecimal,
                aisle: String,
                nutrition: InlineResponse20028Nutrition,
                price: BigDecimal,
                servings: InlineResponse20028Servings,
                spoonacularScore: BigDecimal
                )

object InlineResponse20030 {
    /**
     * Creates the codec for converting InlineResponse20030 from and to JSON.
     */
    implicit val decoder: Decoder[InlineResponse20030] = deriveDecoder
    implicit val encoder: ObjectEncoder[InlineResponse20030] = deriveEncoder
}

File: clojure/src/spoonacular_api/specs/inline_response_200_30.clj
(ns spoonacular-api.specs.inline-response-200-30
  (:require [clojure.spec.alpha :as s]
            [spec-tools.data-spec :as ds]
            [spoonacular-api.specs.inline-response-200-30-ingredients :refer :all]
            [spoonacular-api.specs.inline-response-200-28-nutrition :refer :all]
            [spoonacular-api.specs.inline-response-200-28-servings :refer :all]
            )
  (:import (java.io File)))


(def inline-response-200-30-data
  {
   (ds/req :id) int?
   (ds/req :title) string?
   (ds/req :breadcrumbs) (s/coll-of string?)
   (ds/req :imageType) string?
   (ds/req :badges) (s/coll-of string?)
   (ds/req :importantBadges) (s/coll-of string?)
   (ds/req :ingredientCount) int?
   (ds/opt :generatedText) any?
   (ds/req :ingredientList) string?
   (ds/req :ingredients) (s/coll-of inline-response-200-30-ingredients-spec)
   (ds/req :likes) float?
   (ds/req :aisle) string?
   (ds/req :nutrition) inline-response-200-28-nutrition-spec
   (ds/req :price) float?
   (ds/req :servings) inline-response-200-28-servings-spec
   (ds/req :spoonacularScore) float?
   })

(def inline-response-200-30-spec
  (ds/spec
    {:name ::inline-response-200-30
     :spec inline-response-200-30-data}))
